glusterd2/commands/volumes/common.go

```diff
@@ -41,7 +41,7 @@ func validateOptions(opts map[string]string) error {
 		}
 
 		if err := o.Validate(v); err != nil {
-			return err
+			return fmt.Errorf("Failed to validate value(%s) for key(%s): %s", k, v, err.Error())
```

Comment:
@prashanthpai Shouldn't this be just "err" instead of "err.Error()", as Errorf also accepts interface instead of string

Reply:
Either of it works **in this case**. But as `%s` is specified, and that's a string, I thought it would be explicit to mention `err.Error()` which is also a string.

Further, it won't work for custom error type.

The error type is an interface type. An error variable represents any value that can describe itself as a string. Here is the interface's declaration:

```go
type error interface {
    Error() string
}
```
I wrote a playground example for you to check this out: https://play.golang.org/p/v0uSi8hn1VD

Reply:
@rishubhjain This is a good read: https://blog.golang.org/error-handling-and-go